src/client/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -33,6 +33,12 @@ public static void SetAuthenticationContinuationEventArgs(int requestCode, Resul
             var logger = MsalLogger.Create(Guid.Empty, null);
             logger.Info(string.Format(CultureInfo.InvariantCulture, "Received Activity Result({0})", (int)resultCode));
 
+            if (data == null)
```

Comment:
Having looked more at this, I am not sure this is the correct solution. There are cases where we can interpret the result exclusively from `Result resultCode`, for example for user cancelled.

Reply:
@neha-bhargava Im assuming that line 43 is actually throwing the null ref right? if that is the case, then the code that handles the Result resultCode exclusively (i.e. ProcessFromEmbeddedWebview) will never be reached if there is a null returned. Meaning, there must be some value in data when user cancels otherwise that scenario would throw an exception every time if data is null. 

with that being said I think it might be ok to leave it as is. 

Thoughts @bgavrilMS

Reply:
I taled to @neha-bhargava offline. We can be more defensive and check for data != null. But there are cases where can do better, for example for authentication cancelled we only need the Result.

Reply:
Added logic to include scenarios where data is not required.